### PR TITLE
feat: add settings page to delete llmcache namespace

### PR DESCRIPTION
**Pull request by domeccleston**
<sub><a href="https://huly.app/guest/unkey?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NjZjOGMwNmExZGE1ZmQ5NmY4NDQ2OGEiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctamFtZXMtdW5rZXktNjY2YWYwNzktNjliNzJiMzYyNi05ZGM2MGUiLCJwcm9kdWN0SWQiOiIifQ.Fcr0x0klqyvId-wKjoD7gkW4DapyHMtdXX4ojIXQBY8">Huly&reg;: <b>ENG-1750</b></a></sub>

### CODE REVIEW

File: apps/dashboard/app/(app)/semantic-cache/settings/delete-gateway.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    deleteGateway.mutate({ gatewayId: gateway.id });
+  }
+
+  return (
+    <>
+      <Card className="relative border-2 border-alert">
+        <CardHeader>
+          <CardTitle>Delete</CardTitle>
+          <CardDescription>
+            This gateway will be deleted, along with all of its identifiers and data. This action
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/settings/delete-gateway.tsx:86]: identifiers?

File: apps/dashboard/app/(app)/semantic-cache/settings/delete-gateway.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            Delete gateway
+          </Button>
+        </CardFooter>
+      </Card>
+      <Dialog open={open} onOpenChange={(o) => setOpen(o)}>
+        <DialogContent className="border-alert">
+          <DialogHeader>
+            <DialogTitle>Delete gateway</DialogTitle>
+            <DialogDescription>
+              This gateway will be deleted, along with all of its identifiers and data. This action
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/settings/delete-gateway.tsx:102]: identifiers?

File: apps/dashboard/app/(app)/semantic-cache/settings/page.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Code } from "@/components/ui/code";
+import { getTenantId } from "@/lib/auth";
+import { db } from "@/lib/db";
+import { redirect } from "next/navigation";
+import { DeleteGateway } from "./delete-gateway";
+
+export default async function SemanticCacheSettingPage() {
+  const tenantId = getTenantId();
+  const workspace = await db.query.workspaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/settings/page.tsx:11]: you probably need to load the llm gateway by id
right now you're assuming every user has one at most

File: apps/dashboard/lib/trpc/routers/llmGateway/delete.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    });
+
+    if (!llmGateway || llmGateway.workspace.tenantId !== ctx.tenant.id) {
+      throw new TRPCError({ code: "NOT_FOUND", message: "LLM gateway not found" });
+    }
+
+    await db.transaction(async (tx) => {
+      await tx.delete(schema.llmGateways).where(eq(schema.llmGateways.id, input.gatewayId));
+    });
+
```
Comment (chronark) [inline apps/dashboard/lib/trpc/routers/llmGateway/delete.ts:30]: missing audit log

File: apps/dashboard/app/(app)/semantic-cache/settings/page.tsx (hunk 3 above)
Comment (domeccleston) [reply apps/dashboard/app/(app)/semantic-cache/settings/page.tsx:11]: We should be able to assume that, at the moment it's not possible to create multiple gateways. If the user has already created a gateway, /semantic-cache/new will redirect them

File: apps/dashboard/lib/trpc/routers/llmGateway/delete.ts (hunk 4 above)
Comment (domeccleston) [reply apps/dashboard/lib/trpc/routers/llmGateway/delete.ts:31]: Added

**Review by chronark — APPROVED** [review]

**Pull request merged**